libraries/AP_Math/location.cpp

```diff
@@ -259,6 +259,21 @@ bool location_sanitize(const struct Location &defaultLoc, struct Location &loc)
         loc.alt = defaultLoc.alt;
         has_changed = true;
     }
+
+    // limit lat/lng to appropriate ranges
+    if (abs(loc.lat) > 90 * 1e7 || abs(loc.lng) > 180 * 1e7) {
+        loc.lat = defaultLoc.lat;
+        loc.lng = defaultLoc.lng;
+        has_changed = true;
+    }
+
+    if (abs(loc.alt) >= 0x7fffff) {
```

Comment:
The original goal was to catch saturation/excessively large values, but you're right that its almost never going to catch anything as it wont saturate. Will remove. (I do check at the input for a float as well)